record/custom_records.go

```diff
@@ -22,3 +27,62 @@ func (c CustomSet) Validate() error {
 
 	return nil
 }
+
+// Encode writes the CustomSet to the provided io.Writer in big endian format.
+func (c *CustomSet) Encode(w io.Writer) error {
```

Comment:
We should add unit tests for these encoding/decoding methods as there's usually some kind of surprise malfunction (at least happens to me most of the time) with such things.